flexmeasures/api/v2_0/implementations/sensors.py

```diff
@@ -317,32 +318,36 @@ def post_power_data(
                 ea = parse_entity_address(connection, entity_type="connection")
             except EntityAddressException as eae:
                 return invalid_domain(str(eae))
-            asset_id = ea["sensor_id"]
+            sensor_id = ea["sensor_id"]
 
-            # Look for the Asset object
-            if asset_id in user_asset_ids:
-                asset = Asset.query.filter(Asset.id == asset_id).one_or_none()
+            # Look for the Sensor object
+            if sensor_id in user_sensor_ids:
+                sensor = Sensor.query.filter(Sensor.id == sensor_id).one_or_none()
             else:
                 current_app.logger.warning("Cannot identify connection %s" % connection)
                 return unrecognized_connection_group()
 
             # Validate the sign of the values (following USEF specs with positive consumption and negative production)
-            if asset.is_pure_consumer and any(v < 0 for v in event_values):
+            if sensor.generic_asset.get_attribute("is_pure_consumer") and any(
```

Comment:
So the first condition expects to test a boolean, right? A boolean test on 'false' would succeed:

```
>>> bool("false")
True
```

I notice now that `Sensor.get_attribute` does not type-hint its return value - that would help. I guess it's `Any` (not `str`).

Anyway ― the `db.JSON` column type supports all JSON and here we trust that a boolean has been put in. Correct?

Do we intend to keep these JSON attributes around?
We could consider to define a JSON Schema so such an object can verify itself before being stored. Of course that can be a new issue.

Reply:
I see that PR #246 already addresses the type hint :)

Reply:
I think the JSON field is a nicely extensible structure for holding custom attributes, so I'd like to keep it for now, for some flexibility. But indeed we have no validation in place for these attributes right now. I guess we can expose another registry for plugins to register attribute validators. Just an idea for later, if we need it. But right now, it just gives us flexibility for moving stuff.